File: playground/steps/call_ai.py
config = {
    "name": "Call OpenAI",
    "subscribes": ["call-openai"], 
    "emits": ["openai-response"],
    "input": None,  # No schema validation in Python version
    "workflow": "openai"
}

async def executor(args, emit):
    print('[Call Python OpenAI] Received call_ai event', args);

    await emit({
        "type": "openai-response",
        "data": { "message": args.message },
    });

File: playground/steps/call_ai.rb
class Config
  attr_reader :name, :subscribes, :emits, :input, :flows

  def initialize
    @name = "Call OpenAI"
    @subscribes = ["call-openai"]
    @emits = ["openai-response"]
    @input = nil # No schema validation
    @flows = ["openai"]
  end
end

config = Config.new

def executor(args, emit, ctx)
  ctx.logger.info('[Call Ruby OpenAI] Received call_ai event', args)

  message = args["message"]
  if message.nil?
    ctx.logger.warn('Message not found in args')
    return
  end

  emit.call({
    "type" => "openai-response",
    "data" => { "message" => message }
  })
end
